### PR TITLE
generate flow types

### PR DESCRIPTION
**Pull request by A-gambit**
## Issue #640
Provide flow types, using [typedef-converter](https://github.com/joarwilk/).
Please review, @mweststrate.

### CODE REVIEW

**Comment by mweststrate** [comment]
@A-gambit thanks!

I think the overloads for `observable` are the only pressing lacking here? It would be nice to add a test that invokes the flow compiler against a flow annotated javascript file and test whether it passes. Is that doable?

**Comment by A-gambit** [comment]
@mweststrate  I think yes. I will do it later on this week.

**Comment by A-gambit** [comment]
@mweststrate I review types and fix some bugs.
Unfortunately, flow doesn't work ok with the function which returns different types, like observable. Or action which can get different types of arguments. (https://flowtype.org/docs/functions.html#overloading-caution)

**Comment by mweststrate** [comment]
@A-gambit cool, thanks! So this is ready to be added as first version of flow support?

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "test-browser-safari": "npm run small-build && ( browserify test/*.js -t [ babelify --presets [ es2015 ] ] | tape-run --browser safari )",
     "test-browser-firefox": "npm run small-build && ( browserify test/*.js  | tape-run --browser firefox )",
     "test-travis": "npm run small-build && npm run build-tests && tape test/*.js test/perf/index.js && tsc && istanbul cover tape test/*.js",
+    "test-flow": "node_modules/.bin/flow check",
```
Comment (mweststrate) [inline package.json]: probably should be invoked from `test-travis` and `full-test`

**Comment by A-gambit** [comment]
@mweststrate Sure. I have just add flow check to test scripts.

**Comment by A-gambit** [comment]
@mweststrate So, I can marge?

**Comment by mweststrate** [comment]
yep. should we merge it to just mobx3 branch or also to master? v3 should
come out in the next days

Op ma 28 nov. 2016 13:55 schreef Gregory Shehet <notifications@github.com>:

**Comment by A-gambit** [comment]
@mweststrate Ok, I will merge it to v3, and add new v3 methods to flow types. I think it will be better.

**Comment by A-gambit** [comment]
@mweststrate Done. Can I merge?

**Comment by mweststrate** [comment]
yes! don't forget to update changelog.md. great work, much appreciated!

Op ma 28 nov. 2016 17:04 schreef Gregory Shehet <notifications@github.com>:

**Pull request merged**